src/System.Private.CoreLib/shared/System/Number.NumberBuffer.cs

```diff
@@ -10,7 +11,7 @@ namespace System
     internal static partial class Number
     {
         //  We need 1 additional byte, per length, for the terminating null
-        private const int DecimalNumberBufferLength = 50 + 1;
+        private const int DecimalNumberBufferLength = 30 + 1;  // 29 for the longest input + 1 for rounding
```

Comment:
Possibly write these as 
`private const int DecimalNumberBufferLength = 29 +1 + 1;  // 29 for the longest input + 1 for rounding`
because the +1 you see is not the +1 for rounding...

Reply:
Fixed.